doc/autoinstallation.md

```diff
@@ -0,0 +1,99 @@
+
+There are two ways of give a profile to _AutoYaST_, with (`autoyast` or with
+`autoyast2 parameters`), the principal difference from networking point of 
+view is that `autoyast` will not fetch the profile which means that doesn't
+need the network configuration at all while for `autoyast2` _Linuxrc_ will try
+to configure it if needed.
+
+The current steps that involves network configuration are:
+
+ - _Linuxrc_
+ - First Stage
+   - autoinit
+   - autosetup
+   - finish
+      - network_finish
+        - save_network
+ - Second Stage
+   - autoconfigure
+
+## Linuxrc
+
+The network configuration is basically the same that for
+[installation](installation.md#Linuxrc), but in case that `autoyast2` is used
+then it will fetch and parse the linuxrc options given in the profile and for
+that as commented previously will configure the network if needed.
+
+
+## First Stage
+
+### autoinit
+
+Autoinit will call `iSCSI` or `FCOE` clients if them are enable in _Linuxrc_
+and will try to fetch and process the profile.
+
+### autosetup
+
+This client basically will read the `networking` section in the general one,
+and also will check if `network` requires manual configuration having an
+entry in the 
+[semi-automatic](https://www.suse.com/documentation/sles-12/singlehtml/book_autoyast/book_autoyast.html#CreateProfile.Register)
+section.
+
+It will write the network configuration only if either `setup_before_proposal`
+or `semi-automatic` network configuration has been defined in the profile.
+
+### finish
+
+This client will perform various steps, calling other clients to save the
+final configuration which related to networking area:
+
+  - network_finish
+    - save_network
+
+#### network_finish
+
+This client calls `save_network` which copies udev rules and ifcfg files from
+from the running system which should be the linuxrc config and not the profile
+network configuration. 
+
+Take in account that the copy of the ifcfg files will be done only if 
+`keep_install_network` has not been set to `false` and in case that we need to
+configure the network before _SecondStage_ for registration for example
+`setup_before_proposal` has to be set to `true` as commented in the [autosetup section](#ausosetup), 
+more details about network 
+[here](https://www.suse.com/documentation/sles-12/singlehtml/book_autoyast/book_autoyast.html#CreateProfile.Network)
+
+#### ssh_settings_finish
+
+## Second Stage
+
+The _SecondStage_, also known as `continue` in the control file, is basically 
+the stage where the configuration should be definitly written.
+
+### autoconfigure
+
+This client will read the 
+[desktop configuration files](https://yastgithubio.readthedocs.io/en/latest/autoyast-development/#desktop-configuration-file)
+and is very probably that will write the configuration of our network calling
+the lan_auto client with the information read in the networking section, but
+as commented, it depents on the information of the desktop files to do its job.
+
+
```

Comment:
this client is not part of yast2-network but autoyast itself. Autoyast just generates a list of clients to call. The list is generated using "I-always-forget-option-name" from desktop file(s). So to get to the point, I think that this paragraph is useless and doesn't belong into network's documentation.

lan_auto client is THE client which uses AutoYaST api (Read, Write, Import, Export, ... and other calls) - entry point for AutoYaST either in autoinstallation even in running system.